### PR TITLE
VSTest targeting netstandard 2.0

### PR DESCRIPTION
**Pull request by vagisha-nidhi**
Related issue: [https://github.com/microsoft/vstest/issues/2073](https://github.com/microsoft/vstest/issues/2073)

### CODE REVIEW

File: src/datacollector/datacollector.csproj [hunk 1]
```diff
@@ -9,7 +9,7 @@
   </PropertyGroup>
   <PropertyGroup>
     <AssemblyName>datacollector</AssemblyName>
-    <TargetFrameworks>netcoreapp1.0;netcoreapp2.0;net451</TargetFrameworks>
+    <TargetFrameworks>netcoreapp2.0;net451;netstandard2.0</TargetFrameworks>
```
Comment (cltshivash) [inline src/datacollector/datacollector.csproj]: >netstandard2.0 [](start = 43, length = 14)

is this needed..?

File: scripts/build.ps1 [hunk 2]
```diff
@@ -93,7 +93,6 @@ $TPB_TargetFramework = "net451"
 $TPB_TargetFrameworkCore = "netcoreapp1.0"
```
Comment (cltshivash) [inline scripts/build.ps1]: >$TPB_TargetFrameworkCore [](start = 0, length = 24)

why is this still present ?

**Comment by cltshivash** [comment]
>     }

can we remove this ?

---
Refers to: scripts/build.ps1:170 in f0e0007. [](commit_id = f0e0007270d08c2db55757e5283ab119dc3ac618, deletion_comment = False)

File: src/Microsoft.TestPlatform.Client/Microsoft.TestPlatform.Client.csproj [hunk 3]
```diff
@@ -6,8 +6,8 @@
   <Import Project="$(TestPlatformRoot)scripts/build/TestPlatform.Settings.targets" />
   <PropertyGroup>
     <AssemblyName>Microsoft.VisualStudio.TestPlatform.Client</AssemblyName>
-    <TargetFrameworks>netstandard1.4;net451</TargetFrameworks>
-    <TargetFrameworks Condition=" '$(DotNetBuildFromSource)' == 'true' ">netstandard1.4</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;net451</TargetFrameworks>
+    <TargetFrameworks Condition=" '$(DotNetBuildFromSource)' == 'true' ">netstandard2.0</TargetFrameworks>
```
Comment (cltshivash) [inline src/Microsoft.TestPlatform.Client/Microsoft.TestPlatform.Client.csproj]: can this move into the TestPlatform.settings.targets ? Will prevent adding for every project

**Comment by cltshivash** [comment]
>     <Reference Include="Microsoft.CSharp" />

can we check why this differentiation is needed ?

---
Refers to: src/Microsoft.TestPlatform.CommunicationUtilities/Microsoft.TestPlatform.CommunicationUtilities.csproj:32 in f0e0007. [](commit_id = f0e0007270d08c2db55757e5283ab119dc3ac618, deletion_comment = False)

**Comment by cltshivash** [comment]
>       <Version>4.0.1</Version>

will these versions hold good or needs to be bumped up as well ?

---
Refers to: src/Microsoft.TestPlatform.Extensions.TrxLogger/Microsoft.TestPlatform.Extensions.TrxLogger.csproj:34 in f0e0007. [](commit_id = f0e0007270d08c2db55757e5283ab119dc3ac618, deletion_comment = False)

File: src/Microsoft.TestPlatform.PlatformAbstractions/Microsoft.TestPlatform.PlatformAbstractions.csproj [hunk 4]
```diff
@@ -31,10 +31,10 @@
     <Compile Include="Interfaces\**\*.cs" />
     <Compile Include="Properties\**\*.cs" />
   </ItemGroup>
-  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard1.0' ">
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
     <Compile Include="netstandard1.0\**\*.cs" />
```
Comment (cltshivash) [inline src/Microsoft.TestPlatform.PlatformAbstractions/Microsoft.TestPlatform.PlatformAbstractions.csproj]: >netstandard1.0 [](start = 22, length = 14)

should we rename the folder as just netstandard ?

File: scripts/build.ps1 (hunk 2 above)
Comment (vagisha-nidhi) [reply scripts/build.ps1]: Removed

File: src/Microsoft.TestPlatform.PlatformAbstractions/Microsoft.TestPlatform.PlatformAbstractions.csproj (hunk 4 above)
Comment (vagisha-nidhi) [reply src/Microsoft.TestPlatform.PlatformAbstractions/Microsoft.TestPlatform.PlatformAbstractions.csproj]: Yes. Done

File: scripts/build.sh [hunk 5]
```diff
@@ -120,7 +120,7 @@ DOTNET_CLI_VERSION="LATEST"
 TPB_Solution="TestPlatform.sln"
 TPB_Build_From_Source_Solution="TestPlatform_BuildFromSource.sln"
 TPB_TargetFramework="net451"
-TPB_TargetFrameworkCore="netcoreapp2.0"
+TPB_TargetFrameworkCore="netcoreapp2.1"
 TPB_TargetFrameworkCore10="netcoreapp1.0"
```
Comment (singhsarab) [inline scripts/build.sh]: >TPB_TargetFrameworkCore10 [](start = 0, length = 25)

This will also need removed ?

File: scripts/verify-nupkgs.ps1 [hunk 6]
```diff
@@ -16,10 +16,10 @@ function Verify-Nuget-Packages($packageDirectory)
                      "Microsoft.NET.Test.Sdk" = 13;
                      "Microsoft.TestPlatform" = 421;
                      "Microsoft.TestPlatform.Build" = 19;
-                     "Microsoft.TestPlatform.CLI" = 303;
+                     "Microsoft.TestPlatform.CLI" = 305;
```
Comment (singhsarab) [inline scripts/verify-nupkgs.ps1]: >305 [](start = 52, length = 3)

Can you please mention what is difference here ?

File: scripts/perf/perf.ps1 [hunk 7]
```diff
@@ -11,7 +11,7 @@ Param(
     [System.String] $TargetRuntime = "win7-x64",
 
     [Parameter(Mandatory=$false)]
-    [ValidateSet("netcoreapp1.0", "net451", "net452")]
+    [ValidateSet("net451", "net452")]
```
Comment (singhsarab) [inline scripts/perf/perf.ps1]: Why is this remove and not replace ?

File: src/Microsoft.TestPlatform.CoreUtilities/Microsoft.TestPlatform.CoreUtilities.csproj [hunk 8]
```diff
@@ -20,9 +20,9 @@
     <Reference Include="System" />
     <Reference Include="Microsoft.CSharp" />
   </ItemGroup>
-  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard1.4' ">
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
     <PackageReference Include="System.Diagnostics.FileVersionInfo">
-      <Version>4.0.0</Version>
+      <Version>4.3.0</Version>
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.CoreUtilities/Microsoft.TestPlatform.CoreUtilities.csproj]: Assuming 4.0.0 does not support this ?

File: src/Microsoft.TestPlatform.CrossPlatEngine/Microsoft.TestPlatform.CrossPlatEngine.csproj [hunk 9]
```diff
@@ -25,6 +25,7 @@
     <ProjectReference Include="..\Microsoft.TestPlatform.CoreUtilities\Microsoft.TestPlatform.CoreUtilities.csproj">
       <FromP2P>true</FromP2P>
     </ProjectReference>
+    <PackageReference Include="System.Runtime.CompilerServices.Unsafe" Version="4.5.2" />
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Microsoft.TestPlatform.CrossPlatEngine.csproj]: Explain

File: src/Microsoft.TestPlatform.Extensions.BlameDataCollector/Microsoft.TestPlatform.Extensions.BlameDataCollector.csproj [hunk 10]
```diff
@@ -32,7 +32,7 @@
     <Reference Include="System.Xml" />
     <Reference Include="Microsoft.CSharp" />
   </ItemGroup>
-  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard1.5' ">
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Extensions.BlameDataCollector/Microsoft.TestPlatform.Extensions.BlameDataCollector.csproj]: See if we can get rid of this block.

File: src/Microsoft.TestPlatform.Utilities/Microsoft.TestPlatform.Utilities.csproj [hunk 11]
```diff
@@ -22,6 +22,11 @@
     <Reference Include="System.Xml" />
     <Reference Include="Microsoft.CSharp" />
   </ItemGroup>
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Utilities/Microsoft.TestPlatform.Utilities.csproj]: Why do we need this ?

File: src/package/nuspec/TestPlatform.TestHost.nuspec [hunk 12]
```diff
@@ -14,7 +14,7 @@
     <copyright>© Microsoft Corporation. All rights reserved.</copyright>
     <tags>vstest visual-studio unittest testplatform mstest microsoft test testing</tags>
     <dependencies>
-      <group targetFramework="netcoreapp1.0">
+      <group targetFramework="netstandard2.0">
```
Comment (singhsarab) [inline src/package/nuspec/TestPlatform.TestHost.nuspec]: >netstandard2 [](start = 30, length = 12)

netcoreapp

File: src/package/package/package.csproj [hunk 13]
```diff
@@ -5,8 +5,8 @@
   </PropertyGroup>
   <Import Project="$(TestPlatformRoot)scripts/build/TestPlatform.Settings.targets" />
   <PropertyGroup>
-    <TargetFrameworks>net451;netcoreapp2.0</TargetFrameworks>
-    <TargetFrameworks Condition=" '$(DotNetBuildFromSource)' == 'true' ">netcoreapp2.0</TargetFrameworks>
+    <TargetFrameworks>net451;netcoreapp2.1;netstandard2.0</TargetFrameworks>
```
Comment (singhsarab) [inline src/package/package/package.csproj]: >netstandard2 [](start = 43, length = 12)

remove if not required.

File: src/testhost.x86/testhost.x86.csproj [hunk 14]
```diff
@@ -6,8 +6,8 @@
   <Import Project="$(TestPlatformRoot)scripts/build/TestPlatform.Settings.targets" />
   <PropertyGroup>
     <AssemblyName>testhost.x86</AssemblyName>
-    <TargetFrameworks>netstandard1.4;netcoreapp1.0;net451</TargetFrameworks>
-    <TargetFrameworks Condition=" '$(DotNetBuildFromSource)' == 'true' ">netstandard1.4;netcoreapp1.0</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;netcoreapp2.1;net451</TargetFrameworks>
```
Comment (singhsarab) [inline src/testhost.x86/testhost.x86.csproj]: Do we need both?

**Comment by singhsarab** [comment]
> 

?? Please check this

---
Refers to: test/TestAssets/SimpleTestProjectx86/SimpleTestProjectx86.csproj:12 in 4964d3b. [](commit_id = 4964d3b2e19777489de8afcd2b3444fa4d180229, deletion_comment = False)

File: test/vstest.console.UnitTests/Processors/ResultsDirectoryArgumentProcessorTests.cs [hunk 15]
```diff
@@ -83,30 +83,6 @@ public void InitializeShouldThrowIfArgumentIsAWhiteSpace()
...
-        public void InitializeShouldThrowIfPathIsNotSupported()
-        {
-
-            var folder = @"c:\path\to\in:valid";
-
-            var message = string.Format(
-                @"The path '{0}' specified in the 'ResultsDirectory' is invalid. Error: {1}",
-                folder,
-                "The given path's format is not supported.");
-            this.InitializeExceptionTestTemplate(folder, message);
```
Comment (singhsarab) [inline test/vstest.console.UnitTests/Processors/ResultsDirectoryArgumentProcessorTests.cs]: >InitializeExceptionTestTemplate [](start = 17, length = 31)

Add these back please

File: scripts/verify-nupkgs.ps1 (hunk 6 above)
Comment (vagisha-nidhi) [reply scripts/verify-nupkgs.ps1]: In Microsoft.TestPlatform.CLI
    1. System.Memory.dll is removed(not coming any more)
    2. System.Security.Principal.Windows.dll is coming in these folders.
         a. contentFiles/any/netcoreapp2.1/runtimes/unix/lib/netcoreapp2.0/System.Security.Principal.Windows.dll
         b. contentFiles/any/netcoreapp2.1/runtimes/win/lib/netcoreapp2.0/System.Security.Principal.Windows.dll
         c. contentFiles/any/netcoreapp2.1/System.Security.Principal.Windows.dll

In Microsoft.TestPlatform.ObjectModel
 netstandard1.4 which contained 3 files (CoreUtilities.dll, PlatformAbstractions.dll, ObjectModel.dll) has been removed

In Microsoft.TestPlatform.Portable, these have been removed
1. tools/netcoreapp2.1/System.Memory.dll
2. tools/netcoreapp2.1/TestHost/System.Runtime.InteropServices.RuntimeInformation.dll

File: test/vstest.console.UnitTests/Processors/ResultsDirectoryArgumentProcessorTests.cs (hunk 15 above)
Comment (vagisha-nidhi) [reply test/vstest.console.UnitTests/Processors/ResultsDirectoryArgumentProcessorTests.cs]: Added.

File: src/testhost.x86/testhost.x86.csproj (hunk 14 above)
Comment (vagisha-nidhi) [reply src/testhost.x86/testhost.x86.csproj]: Just targeting netcoreapp2.1 now

File: src/package/package/package.csproj (hunk 13 above)
Comment (vagisha-nidhi) [reply src/package/package/package.csproj]: Removed.

File: src/package/nuspec/TestPlatform.TestHost.nuspec (hunk 12 above)
Comment (vagisha-nidhi) [reply src/package/nuspec/TestPlatform.TestHost.nuspec]: Done

File: src/Microsoft.TestPlatform.Utilities/Microsoft.TestPlatform.Utilities.csproj (hunk 11 above)
Comment (vagisha-nidhi) [reply src/Microsoft.TestPlatform.Utilities/Microsoft.TestPlatform.Utilities.csproj]: Removed.

File: src/Microsoft.TestPlatform.Extensions.BlameDataCollector/Microsoft.TestPlatform.Extensions.BlameDataCollector.csproj (hunk 10 above)
Comment (vagisha-nidhi) [reply src/Microsoft.TestPlatform.Extensions.BlameDataCollector/Microsoft.TestPlatform.Extensions.BlameDataCollector.csproj]: Removed the entire block. Not needed.

File: scripts/perf/perf.ps1 (hunk 7 above)
Comment (vagisha-nidhi) [reply scripts/perf/perf.ps1]: Replaced.

File: src/Microsoft.TestPlatform.CrossPlatEngine/Microsoft.TestPlatform.CrossPlatEngine.csproj (hunk 9 above)
Comment (vagisha-nidhi) [reply src/Microsoft.TestPlatform.CrossPlatEngine/Microsoft.TestPlatform.CrossPlatEngine.csproj]: Removed from here but needed in TestHostProvider.csproj

File: scripts/build.ps1 [hunk 16]
```diff
@@ -227,15 +215,14 @@ function Publish-Package
     Write-Log "Publish-Package: Started."
     $dotnetExe = Get-DotNetPath
     $fullCLRPackageDir = Get-FullCLRPackageDirectory
-    $coreCLRPackageDir = Get-CoreCLRPackageDirectory
```
Comment (vagisha-nidhi) [inline scripts/build.ps1]: >Get-CoreCLRPackageDirectory [](start = 25, length = 27)

Remove this

File: src/Microsoft.TestPlatform.ObjectModel/Microsoft.TestPlatform.ObjectModel.csproj [hunk 17]
```diff
@@ -31,31 +31,8 @@
...
-    </PackageReference>
-    <PackageReference Include="System.ComponentModel.TypeConverter">
-      <Version>4.1.0</Version>
-    </PackageReference>
-    <PackageReference Include="System.Runtime.Serialization.Json">
-      <Version>4.0.2</Version>
-    </PackageReference>
-    <PackageReference Include="System.Reflection.TypeExtensions">
-      <Version>4.1.0</Version>
+      <Version>1.6.0</Version>
```
Comment (vagisha-nidhi) [inline src/Microsoft.TestPlatform.ObjectModel/Microsoft.TestPlatform.ObjectModel.csproj]: >1.6.0 [](start = 15, length = 5)

Do we bump up this version or remain at 1.3.0?

**Comment by vagisha-nidhi** [comment]
>     <Reference Include="Microsoft.CSharp" />

Check if needed

---
Refers to: src/Microsoft.TestPlatform.TestHostProvider/Microsoft.TestPlatform.TestHostProvider.csproj:35 in ba48652. [](commit_id = ba4865239d8eae883f3b072a14807199de8ea07b, deletion_comment = True)

File: src/package/nuspec/TestPlatform.ObjectModel.nuspec [hunk 18]
```diff
@@ -16,29 +16,10 @@
...
-        <dependency id="System.Diagnostics.Process" version="[4.1.0, )" />
-        <dependency id="System.Diagnostics.TextWriterTraceListener" version="[4.0.0, )" />
-        <dependency id="System.Diagnostics.TraceSource" version="[4.0.0, )" />
-        <dependency id="System.Threading.Thread" version="[4.0.0, )" />
-      </group>
-      <group targetFramework="netstandard1.4">
-        <dependency id="System.ComponentModel.TypeConverter" version="[4.1.0, )" />
-        <dependency id="System.Runtime.InteropServices.RuntimeInformation" version="[4.0.0, )" />
+      <group targetFramework="netstandard2.0">
+        <dependency id="System.Xml.XPath.XmlDocument" version="[4.3.0, )" />
```
Comment (vagisha-nidhi) [inline src/package/nuspec/TestPlatform.ObjectModel.nuspec]: ><dependency id="System.Xml.XPath.XmlDocument" version="[4.3.0, )" /> [](start = 8, length = 68)

Remove this

File: src/package/nuspec/TestPlatform.TranslationLayer.nuspec [hunk 19]
```diff
@@ -18,19 +18,9 @@
...
-        <dependency id="System.Xml.XPath.XmlDocument" version="[4.0.1, )" />
-
-        <!-- CommunicationUtilities dependencies are covered above -->
-        <!-- TestPlatform.Common dependencies -->
-        <dependency id="System.Runtime.Loader" version="[4.0.0, )" />
-        <dependency id="Microsoft.TestPlatform.ObjectModel" version="$Version$"/>
-        <dependency id="Newtonsoft.Json" version="$JsonNetVersion$"/>
+      <group targetFramework="netstandard2.0">
+        <dependency id="NETStandard.Library" version="[2.0.0, )" />
+        <dependency id="System.Xml.XPath.XmlDocument" version="[4.3.0, )" />
```
Comment (vagisha-nidhi) [inline src/package/nuspec/TestPlatform.TranslationLayer.nuspec]: ><dependency id="System.Xml.XPath.XmlDocument" version="[4.3.0, )" /> [](start = 8, length = 68)

Remove

File: src/vstest.console/Processors/ResultsDirectoryArgumentProcessor.cs [hunk 20]
```diff
@@ -140,7 +140,13 @@ public void Initialize(string argument)
             try
             {
                 // to check valid directory path
-                var di = new DirectoryInfo(argument);
+                char[] invalidPathChars = Path.GetInvalidPathChars();
```
Comment (vagisha-nidhi) [inline src/vstest.console/Processors/ResultsDirectoryArgumentProcessor.cs]: >invalidPathChars [](start = 23, length = 16)

Not needed

File: test/Microsoft.TestPlatform.TestUtilities/Microsoft.TestPlatform.TestUtilities.csproj [hunk 21]
```diff
@@ -21,9 +21,9 @@
       <Version>$(MSTestFrameworkVersion)</Version>
     </PackageReference>
   </ItemGroup>
-  <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp1.0' ">
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp2.1' ">
     <PackageReference Include="System.Xml.XPath.XmlDocument">
-      <Version>4.0.1</Version>
+      <Version>4.3.0</Version>
     </PackageReference>
```
Comment (vagisha-nidhi) [inline test/Microsoft.TestPlatform.TestUtilities/Microsoft.TestPlatform.TestUtilities.csproj]: Check if required

File: scripts/build.ps1 (hunk 16 above)
Comment (vagisha-nidhi) [reply scripts/build.ps1]: removed

File: src/Microsoft.TestPlatform.ObjectModel/Microsoft.TestPlatform.ObjectModel.csproj (hunk 17 above)
Comment (vagisha-nidhi) [reply src/Microsoft.TestPlatform.ObjectModel/Microsoft.TestPlatform.ObjectModel.csproj]: @singhsarab

Comment (vagisha-nidhi) [reply src/Microsoft.TestPlatform.ObjectModel/Microsoft.TestPlatform.ObjectModel.csproj]: > > �
> 
> ?? Please check this
> 
> Refers to: test/TestAssets/SimpleTestProjectx86/SimpleTestProjectx86.csproj:12 in [4964d3b](https://github.com/microsoft/vstest/commit/4964d3b2e19777489de8afcd2b3444fa4d180229). [](commit_id = [4964d3b](https://github.com/microsoft/vstest/commit/4964d3b2e19777489de8afcd2b3444fa4d180229), deletion_comment = False)

Seems normal when opened in VS.

**Comment by vagisha-nidhi** [comment]
> > �
> 
> ?? Please check this
> 
> Refers to: test/TestAssets/SimpleTestProjectx86/SimpleTestProjectx86.csproj:12 in [4964d3b](https://github.com/microsoft/vstest/commit/4964d3b2e19777489de8afcd2b3444fa4d180229). [](commit_id = [4964d3b](https://github.com/microsoft/vstest/commit/4964d3b2e19777489de8afcd2b3444fa4d180229), deletion_comment = False)

Seems normal when opened in VS.

**Comment by vagisha-nidhi** [comment]
> > ```
> > <Reference Include="Microsoft.CSharp" />
> > ```
> 
> Check if needed
> 
> Refers to: src/Microsoft.TestPlatform.TestHostProvider/Microsoft.TestPlatform.TestHostProvider.csproj:35 in [ba48652](https://github.com/microsoft/vstest/commit/ba4865239d8eae883f3b072a14807199de8ea07b). [](commit_id = [ba48652](https://github.com/microsoft/vstest/commit/ba4865239d8eae883f3b072a14807199de8ea07b), deletion_comment = True)

Will take this up with net451 changes

File: src/Microsoft.TestPlatform.ObjectModel/Microsoft.TestPlatform.ObjectModel.csproj (hunk 17 above)
Comment (singhsarab) [reply src/Microsoft.TestPlatform.ObjectModel/Microsoft.TestPlatform.ObjectModel.csproj]: Keep it at 1.3.0 if it has netstandard2.0 support

---
In reply to: [301028228](https://github.com/microsoft/vstest/pull/2067#discussion_r301028228) [](ancestors = 301028228)

File: src/vstest.console/Processors/ResultsDirectoryArgumentProcessor.cs [hunk 22]
```diff
@@ -140,7 +140,12 @@ public void Initialize(string argument)
             try
             {
                 // to check valid directory path
-                var di = new DirectoryInfo(argument);
+                if (!string.IsNullOrEmpty(argument) && !IsValidPath(argument) )
+                {
+                    throw new ArgumentException("Illegal characters in path.");
+                }
+
+                var di = new FileInfo(argument);
```
Comment (singhsarab) [inline src/vstest.console/Processors/ResultsDirectoryArgumentProcessor.cs]: >FileInfo [](start = 29, length = 8)

Keep it DirectoryInfo

**Review by singhsarab — APPROVED** [review]
Approving with suggestions.

Please do make sure you have made all the changes requested. Removed netstandard1.4 references from test assets as well.

**Comment by sharwell** [comment]
This is super unfortunate. We have projects targeting **netstandard1.x** (test running on **netcoreapp1.1**), and this change means we can never update Microsoft.NET.Test.Sdk to any version newer than 16.1.1.

**Comment by nohwnd** [comment]
@sharwell Is it not possible to update to netcoreapp2.1, or any of the supported .NET Core versions?

**Comment by sharwell** [comment]
@nohwnd We use netcoreapp2.1 for testing our netstandard2.0 build.

**Comment by nohwnd** [comment]
I am afraid we can't help you there 🙁 I suspect downgrading testhost to netcoreapp1.1 would be a significant effort. /cc @pavelhorak

**Pull request merged**